QRCoder/PayloadGenerator.cs

```diff
@@ -1552,7 +1557,7 @@ public override string ToString()
                                 bezahlCodePayload += $"creditorid={ Uri.EscapeDataString(creditorId)}&";
                             if (!string.IsNullOrEmpty(mandateId))
                                 bezahlCodePayload += $"mandateid={ Uri.EscapeDataString(mandateId)}&";
-                            if (dateOfSignature != null)
+                            if (dateOfSignature != DateTime.MinValue)
```

Comment:
DateTime is a struct. It cannot be null. Also a warning appeared for this after add of .NET 5.

Reply:
Thanks for the feedback. The documentation states that for nullable types both, checking for null and checking for "HasValue" works: https://docs.microsoft.com/de-de/dotnet/csharp/language-reference/builtin-types/nullable-value-types
(Check second and third example.)

So I suggest we change the code to:
```suggestion
                            if (dateOfSignature.HasValue && dateOfSignature >= DateTime.MinValue)
```

Reply:
I guess we have another issue here. The constructor does already convert the Nullable type to a regulare DateTime. In this case, nothing is assign and it will be the default value, which is probably `DateTime.MinValue`. This means there is no relation between your comment and this compare.

Reply:
> The constructor does already convert the Nullable type to a regulare DateTime.
But it does so, only if dateOfSignature != null otherwise the conversion isn't run.